### ISSUE TITLE
Is there any way to customize the section styling?

### ISSUE THREAD

**Issue by bdrelling** [xmartlabs/Eureka#266]
I couldn't find any documentation on this, but essentially we're trying something simple like adding 10px of margin to either side of our form and have 2px corner radius on every form section.

I saw this thread, but it went unresolved: https://github.com/xmartlabs/Eureka/issues/22

I know custom rows can help me make specific rows, but is there any way to just add this gap between the controller view edges and the form and round a section similar to how it would look on the iPad?

**Comment by mats-claassen** [comment]
Hi @bdrelling, to have a cornerRadius defined on every section of the form you will have to use views for those section headers and add a cornerRadius there. If you use the same class of view then you have to do it only once.
To get a margin to either side of the form, you could change the constraints or margins of the tableView.
Is this what you want?:
![captura de pantalla 2016-02-18 a las 2 48 20 p m](https://cloud.githubusercontent.com/assets/8573865/13152558/8c799b3e-d64e-11e5-82a7-1a75003103eb.png)

What I did was just to set the `leading` and `trailing` constraints of the tableView to 10 to get the margins. And in the header view I added a cornerRadius and a plain view with the same backgroundColor to hide the bottom corners.

If you want rounded corners at the bottom of the section then you will have to create a custom row or use a footer view that simulates those rounded corners.

Tell me if this was helpful

**Comment by bdrelling** [comment]
@mats-claassen Thanks for reaching out. 

To provide a clearer example, this is the sort of form I'd like to be able to create using Eureka so all of our forms are powered by the same logic and library:

![img_8123](https://cloud.githubusercontent.com/assets/3022693/13156525/e100d850-d650-11e5-8d7b-7d48e7601dda.PNG)

I'll try this within the next few days and see if I can achieve the effect I'm looking for.

**Comment by mats-claassen** [comment]
I believe it should be possible by setting up a tableView (in the storyboard or not) with those constraints to the margin and set a corner radius to the tableView.

**Comment by bdrelling** [comment]
@mats-claassen I didn't see any documentation as to how I can set up Eureka in a storyboard or relate it to a specific tableView. If I could do that, that would be perfect.

**Comment by mats-claassen** [comment]
Have you had a look at the examples project? There is the `Main.storyboard` with a lot of view controllers with a table view. Normally we set the constraints of the table views to the border but you can change that.

**Comment by bdrelling** [comment]
@mats-claassen I did not see the Main.storyboard, did not even think to look in there. Sorry for the oversight and thanks for the patience! Will update this with results.

**Comment by bdrelling** [comment]
@mats-claassen I loaded up the Examples project and manipulated the `tableView` to get what I wanted it to look like. Then, I tried to apply the same steps to my project, but I can't get it to look anything other than the default.

My code:

    class EurekaTestController: FormViewController {
	    override func viewDidLoad() {
		    super.viewDidLoad()

		    form

		    +++ Section("Dates")
		    <<< TextRow() {
	   		    $0.title = "Start Date"
		    }
		    <<< TextRow() {
		   	    $0.title = "Start Date"
		    }
	    }
    }

My Interface Builder:

![img_8123](https://cloud.githubusercontent.com/assets/3022693/13194671/f92f0cb8-d767-11e5-8339-71f879b73d90.png)
![img_8124](https://cloud.githubusercontent.com/assets/3022693/13194670/f92e70aa-d767-11e5-9830-69ad535e05b0.png)

Result Screenshots:

![img_8125](https://cloud.githubusercontent.com/assets/3022693/13194677/53a1755a-d768-11e5-8bc2-fa818ce094cb.PNG)

**Comment by mats-claassen** [comment]
@bdrelling You have some errors in the interface builder (red icon next to `Eureka Test Controller Scene`). 
From what I see your constraints seem ok, but you need to add a constraint for the bottom of the tableView. 

And **do not forget** connecting the tableViews **outlet** to your FormViewController. (to the property that is in `FormViewController`). If you don't, Eureka will create the tableView programmatically and I guess that is happening.

**Comment by bdrelling** [comment]
@mats-claassen Requiring a constraint for the bottom of the `tableView` could be an issue if we want it to be dynamic. In my example above, we are using 7 fields. We might not always know how many fields we want to present on any given form in order to keep it dynamic. If we have to set a constraint for the bottom of the `tableView`, it can't grow with a conditional number of rows unless we count up the rows and do the math programmatically. Was hoping to avoid something like this. Is the bottom constraint necessary in this case?

I did not connect the outlet to the controller this way, was not sure this was what had to happen. Thanks for letting me know about this! That also taught me a lot about some interesting patterns I can work with, should be fun to experiment with.

I had one other question regarding the storyboards: If I want a static view fixed to the bottom of my controller. I put a TableView into my controller via the storyboard and laid it out properly such that my FixedView was on top. I added a dropshadow to the FixedView. Something in Eureka's code sends the TableView to the front, so I have to manually send it to back in viewDidLoad to get it work. Is something causing this that we can identify so I can remove the workaround, or is this intentional?

**Comment by mats-claassen** [comment]
Well, regarding the bottom constraint on the table: it might be that you don't need it but it will most probably not behave well.

ON the static view at the bottom, I have done that from code, adding it at the bottom with constraints and it worked fine. I have also once added the view at the bottom and made the tableview not to go to the end of the screen but where the bottom view starts. But these are just workarounds.

 I think there is no code in Eureka that brings the tableView to front. Have you used the view inspector of Xcode? If you did connect the outlet correctly then it should not happen

**Comment by bdrelling** [comment]
I spent some time trying to figure it out but couldn't narrow it down. When I used a UIView in the same place that I had the TableView, the drop shadow was visible. When I removed the UIView and connected the TableView via the outlet, the drop shadow was hidden. As you said, I had it so the bottom constraint of the TableView was connected to the top constraint of the FixedView.

This was all fixed if I did `sendSubviewToBack` in the `viewDidLoad` method of my controller.

I have it working now and I think I have a full grasp on storyboard issues. Is there anything I can do to help troubleshoot this issue, or is it not worth looking into further?

Also, thanks for all of your help and your quick replies. You guys are amazing! Eureka will make up 95% of our Point of Sale app, so I'm excited to see it has a lot of support.
